File: ziteng/Longest Increasing Continuous subsequence.java
public class Solution {
    /**
     * @param A an array of Integer
     * @return  an integer
     */
    public int longestIncreasingContinuousSubsequence(int[] A) {
        if(A.length==0) return 0;
        int res = 0;
        int temp = 0;
        int up = 0;
        int down = 0;
        for(int i = 1; i < A.length; i++) {
            if(A[i]>A[i-1]) {
                up++;
                down = 0;
            }
            if(A[i]<A[i-1]) {
                down++;
                up = 0;
            }
            temp = Math.max(up,down);
            res = Math.max(temp,res);
        }
        return res+1;
    }
}

File: mu_wang/9_13/Longest Increasing Continuous subsequence.py
class Solution:
    # @param {int[]} A an array of Integer
    # @return {int}  an integer
    def longestIncreasingContinuousSubsequence(self, A):
        # Write your code here
        asc = 1
        desc = 1
        maxAsc = 1
        maxDesc = 1
        
        if len(A)<1:
            return 0
        
        for i in range(1,len(A)):
            if  A[i-1] < A[i]:
                desc = 1
                asc += 1
                maxAsc = max(asc,maxAsc)
            if A[i-1] > A[i]:
                asc = 1
                desc += 1
                maxDesc = max(maxDesc,desc)
            
        maxLen = max(maxAsc,maxDesc)
        return maxLen  
   